pkg/webui/console/views/device-overview/index.js

```diff
@@ -219,8 +241,17 @@ class DeviceOverview extends React.Component {
   }
 
   render() {
-    const { device } = this.props
+    const { device, shouldRedirect } = this.props
     const devIds = device && device.ids
+
+    if (shouldRedirect) {
+      toast({
+        type: toast.types.WARNING,
+        message: m.failedAccessOtherHostDeviceMessage,
+      })
+      return <Redirect to="/applications" />
+    }
```

Comment:
I think we should rather solve this with the `withFeatureRequirement()` HOC in `console/lib/components/with-feature-requirement.js`. This will allow us to better control redirect behavior as a whole.